math/mathcore/src/Util.cxx

```diff
@@ -0,0 +1,69 @@
+#include "Math/Util.h"
+
+#include <chrono>
+
+ROOT::Math::Util::TimingScope::TimingScope(std::function<void(std::string const &)> printer, std::string const &message)
+   : fPrinter{printer}, fMessage{message}
+{
+   using std::chrono::steady_clock;
+   fBegin = new steady_clock::time_point{steady_clock::now()};
```

Comment:
Why are we 
a) heap-allocating a `time_point`
b) storing it as a `void*` and
c) reinterpret-casting it back?

This seems weird and inefficient...

Reply:
Why is `<chrono>` less acceptable than, say `<functional>`? It seems a pretty innocent include to me

Reply:
`<functional>` is used in other mathcore headers already. It's just that recently I got some flak for being the first to include `<any>` in some other part of ROOT and I wanted to avoid discussions about that. I agree with you that's not a well-founded argument. Let me change this PR to include `<chrono>` in the header and if @lmoneta has a problem with that he can still comment.